test/unit/deadline/test_resources.py

```diff
@@ -1061,63 +1129,44 @@ def test_assert_single_task_log_contains_success(self, job: Job, session: Sessio
         step.list_tasks.assert_called_once_with(deadline_client=deadline_client)
         task.get_last_session.assert_called_once_with(deadline_client=deadline_client)
 
-    def test_assert_single_task_log_contains_multi_step(self, job: Job) -> None:
-        # GIVEN
-        deadline_client = MagicMock()
-        logs_client = MagicMock()
-        step = MagicMock()
-        expected_pattern = re.compile(r"a message")
-
-        with (patch.object(job, "list_steps", return_value=[step, step]) as mock_list_steps,):
-
-            # WHEN
-            def when():
-                job.assert_single_task_log_contains(
-                    deadline_client=deadline_client,
-                    logs_client=logs_client,
-                    expected_pattern=expected_pattern,
-                )
-
-            # THEN
-            with pytest.raises(AssertionError) as raise_ctx:
-                when()
-
-        print(raise_ctx.value)
-
-        assert raise_ctx.match("Job contains multiple steps")
-        mock_list_steps.assert_called_once_with(deadline_client=deadline_client)
-        step.list_tasks.assert_not_called()
-
-    def test_assert_single_task_log_contains_multi_task(self, job: Job, session: Session) -> None:
+    def test_assert_single_task_log_does_not_contain_success(
+        self, job: Job, session: Session
+    ) -> None:
         # GIVEN
         deadline_client = MagicMock()
         logs_client = MagicMock()
         step = MagicMock()
         task = MagicMock()
-        step.list_tasks.return_value = [task, task]
+        step.list_tasks.return_value = [task]
         task.get_last_session.return_value = session
         expected_pattern = re.compile(r"a message")
 
-        with (patch.object(job, "list_steps", return_value=[step]) as mock_list_steps,):
+        with (
+            patch.object(job, "list_steps", return_value=[step]) as mock_list_steps,
+            patch.object(
+                session, "assert_log_does_not_contain"
+            ) as mock_session_assert_log_does_not_contain,
+        ):
 
             # WHEN
-            def when():
-                job.assert_single_task_log_contains(
-                    deadline_client=deadline_client,
-                    logs_client=logs_client,
-                    expected_pattern=expected_pattern,
-                )
-
-            # THEN
-            with pytest.raises(AssertionError) as raise_ctx:
-                when()
-
-        print(raise_ctx.value)
+            job.assert_single_task_log_does_not_contain(
+                deadline_client=deadline_client,
+                logs_client=logs_client,
+                expected_pattern=expected_pattern,
+            )
 
-        assert raise_ctx.match("Job contains multiple tasks")
+        # THEN
+        # This test is only to confirm that no assertion is raised, since the expected message
+        # is in the logs
+        mock_session_assert_log_does_not_contain.assert_called_once_with(
```

Comment:
Sorry no I don't have an additional insight. I saw that the positive assertion method had this test so I replicated it for my negative assertion test case https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/mainline/test/unit/deadline/test_resources.py#L1028-L1062

Reply:
I ended up chatting with @jusiskin who wrote the original tests. It seems that this was intended to be a unit test for the utility function and ensures that it calls the tested function underneath. Not something we have to change right now, but this may make more sense as an integration test to verify it's actually doing what we want. Resolving!